cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -148,12 +152,15 @@ func (p *Planner) NodesToDelete() (empty, needDrain []*apiv1.Node) {
 		append(emptyRemovable, needDrainRemovable...),
 		p.context.AutoscalingOptions.MaxScaleDownParallelism)
 	for _, nodeToRemove := range nodesToRemove {
-		if len(nodeToRemove.PodsToReschedule) > 0 {
+		if len(nodeToRemove.PodsToReschedule) > 0 && !nodeToRemove.IsRisky {
 			needDrain = append(needDrain, nodeToRemove.Node)
+		} else if len(nodeToRemove.PodsToReschedule) > 0 && nodeToRemove.IsRisky {
```

Comment:
I think it would be more readable to have nested `if` instead of repeating the condition for whether the node needs drain:

```
if len(nodeToRemove.PodsToReschedule) > 0 {
    if nodeToRemove.IsRisky {
        (...)
    } else {
        (...)
    }
} else {
    (...)
}
```